### PR TITLE
Add IOFunctions tests

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

### CODE REVIEW

File: core/src/test/java/fj/data/IOFunctionsTest.java [hunk 1]
```diff
@@ -56,9 +57,55 @@ public void close() {
...
+
+  @Test
+  public void testSequenceWhile() throws IOException {
+    BufferedReader r = new BufferedReader(new StringReader("foo1\nbar2\nfoobar3"));
+    Stream<IO<String>> s1 = Stream.repeat(() -> r.readLine());
+    IO<Stream<String>> io = sequenceWhile(s1, s -> !s.equals("foobar3"));
+    ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    PrintStream originalOut = System.out;
+    System.setOut(new PrintStream(outContent));
+    runSafe(io).foreach(s -> runSafe(stdoutPrint(s)));
```
Comment (jbgi) [inline core/src/test/java/fj/data/IOFunctionsTest.java]: the test is named `testSequenceWhile` but `foreach` is what is actually tested?

File: core/src/test/java/fj/data/IOFunctionsTest.java [hunk 2]
```diff
@@ -56,9 +57,55 @@ public void close() {
...
+
+  @Test
+  public void testReplicateM() throws IOException {
+    final IO<String> is = () -> new BufferedReader(new StringReader("foo")).readLine();
+    assertThat(replicateM(is, 3).run(), is(List.list("foo", "foo", "foo")));
+  }
+
+
+  @Test
+  public void testLift() throws IOException {
```
Comment (jbgi) [inline core/src/test/java/fj/data/IOFunctionsTest.java]: is `lift` tested?

File: core/src/test/java/fj/data/IOFunctionsTest.java (hunk 1 above)
Comment (gliptak) [reply core/src/test/java/fj/data/IOFunctionsTest.java]: there is sequenceWhile above. I could split to two tests ...

File: core/src/test/java/fj/data/IOFunctionsTest.java (hunk 2 above)
Comment (gliptak) [reply core/src/test/java/fj/data/IOFunctionsTest.java]: just moved this code from the other Test.java I deleted

**Pull request merged**